test/object-store/util/test_file.cpp

```diff
@@ -342,6 +361,7 @@ TestSyncManager::TestSyncManager(const Config& config, const SyncServer::Config&
     sc_config.base_file_path = m_base_file_path;
     sc_config.metadata_mode = config.metadata_mode;
     sc_config.log_level = config.verbose_sync_client_logging ? util::Logger::Level::all : util::Logger::Level::off;
+    sc_config.socket_factory = util::websocket::SocketFactory::defaultSocketFactory;
```

Comment:
instead of adding a new constructor above, can you just add the socket factory factory as a member of TestSyncManager::Config and handle the special case of overriding it in the constructor on line 331?